templates/components/products/schema.html

```diff
@@ -0,0 +1,61 @@
+<script type="application/ld+json">
+    {
+        "@context": "https://schema.org/",
+        "@type": "Product",
+        "name": {{{JSONstringify product.title}}},
+        {{#if product.sku}}"sku": "{{product.sku}}",{{/if}}
+        {{#if product.mpn}}"mpn": "{{product.mpn}}",{{/if}}
+        {{#if product.gtin}}"gtin{{length product.gtin}}": "{{product.gtin}}",{{/if}}
+        "url" : "{{product.url}}",
+        {{#if product.brand}}
+        "brand": {
+            "@type": "Brand",
+            "url": "{{product.brand.url}}",
+            "name": {{{JSONstringify product.brand.name}}}
+        },
+        {{/if}}
+        "description": "{{encodeURI (sanitize product.description)}}",
+        "image": "{{getImage product.main_image 'zoom_size' (cdn theme_settings.default_image_product)}}",
+        {{#and settings.show_product_reviews product.reviews.list.length}}
+        "aggregateRating": {
+            "@type": "AggregateRating",
+            "ratingValue": "{{product.rating}}",
+            "reviewCount": "{{product.num_reviews}}"
+        },
+        "review": [
+        {{#each product.reviews.list}}
+            {
+                "@type": "Review",
+                "author": {
+                    "@type": "Person",
+                    "name": {{{JSONstringify name}}}
+                },
+                "datePublished": "{{date}}",
+                "reviewBody": {{{JSONstringify text}}},
+                "name": {{{JSONstringify title}}},
+                "reviewRating": {
+                    "@type": "Rating",
+                    "bestRating": "5",
+                    "ratingValue": "{{rating}}",
+                    "worstRating": "1"
+                }
+            }{{#unless @last}},{{/unless}}
+        {{/each}}
+        ],
+        {{/and}}
+        "offers": {
+            "@type": "Offer",
+            "priceCurrency": "{{currency_selector.active_currency_code}}",
+            {{#if product.price.price_range}}
+            "minPrice": "{{#if product.price.price_range.min.with_tax}}{{product.price.price_range.min.with_tax.value}}{{else}}{{product.price.price_range.min.without_tax.value}}{{/if}}",
+            "maxPrice": "{{#if product.price.price_range.max.with_tax}}{{product.price.price_range.max.with_tax.value}}{{else}}{{product.price.price_range.max.without_tax.value}}{{/if}}",
+            {{else}}
+            "price": "{{#if product.price.with_tax}}{{product.price.with_tax.value}}{{else}}{{product.price.without_tax.value}}{{/if}}",
```

Comment:
Hey @mattcoy-arcticleaf 👋 
Looks like `minPrice` and `maxPrice` fields are not recognized by [schema.org](https://validator.schema.org/?hl=en-US#url=https%3A%2F%2Fyurytutstore.mybigcommerce.com%2Fsport%2Ffootball-shorts%2F) They may be placed in `priceSpecification` inside `offers`
I am thinking about creating the pr with the fix, what do you think? Or I am missing something and everything is correct? 🤔
Also according to [Rich Results Test](https://search.google.com/test/rich-results/result/r%2Fmerchant-listings?id=MPIhxH1oSr8Xz-47ql8ybA) `price` field should be represented even with price range

Reply:
Google currently has no documented features based around minPrice and maxPrice. They do support AggregateOffer however they consider that as a set of prices from different merchants, like on an affiliate site. 

https://developers.google.com/search/docs/appearance/structured-data/product#merchant-listings_offer 

Maybe Google will add support if more sites use it :-)

Reply:
@mattcoy-arcticleaf thanks for clarification and contribution 👍 
@Tiggerito Google totally fine with `minPrice` and `maxPrice` on `offers` level, but https://schema.org/ validator does not recognize those fields unfortunately. So from my side it is better to have them inside `priceSpecification` to satisfy both validators

Reply:
@Tiggerito @yurytut1993 ,

Realistically, the correct way to display a range of prices for different variants is to make an offers object and specify the price (and other data) of each. However, with BigCommerce, this is only possible through using GraphQL, since the variant data is not available via the Stencil context. I've actually custom-coded this for a few sites, but didn't want to add it to Cornerstone for a number of reasons.

See the LD+JSON tag with offers here: https://www.fishusa.com/Berkley-Deep-Hit-Stick-Premium-Lure/

Reply:
Cool. I found an example that is showing a price range: https://www.fishusa.com/McFly-Foam/

Google does not support variants in a usable way at the moment (They say you have to have a unique URL for each one). They are looking into improving that.

The multiple offer looks to be a good undocumented workaround to get the price range without affecting merchant listings by using aggregateOffer.

Reply:
> @Tiggerito Google totally fine with `minPrice` and `maxPrice` on `offers` level, but https://schema.org/ validator does not recognize those fields unfortunately. So from my side it is better to have them inside `priceSpecification` to satisfy both validators

The Rich Result Test tool focuses on errors and warnings with markup used to get rich results within Google search. It ignores properties it does not care about. While the schema.org validator is a lot stricter with the spec saying those properties have to be in a PriceSpecification.

Adding `minPrice` and `maxPrice` to the PriceSpecification is valid, but at the moment it does not look like Google looks at it.

Reply:
@Tiggerito , the code I wrote on fishUSA provides a unique URL to each variant using the sku parameter. Does it actually work on that site? Tough to say. It would, however, work on a standard BC site since going to that URL would select the correct options. I believe this satisfies Google "unique URL" requirement.

Reply:
@mattcoy-arcticleaf I see several issues:

The variant URLs are canonicalised out via the canonical tag, so they will not be in the index or show in search.

Even if they are not canonicalised out, Google will most likely consider them duplicates of the main product page, and canonicalise them out anyhow.

Typically, there are no real links to the variant pages, making it unlikely that search engines will find or consider the URLs worth indexing. I see you provide links via the structured data, I suspect that is not enough of a signal for Google to decide to index them, especially when they have not documented support for URLs in offers.

This is why I think their previous solution for variants was unworkable on most e-commerce websites.

They've currently removed most of their documentation that defined how to mark up variants. All that is left is the requirement that they have their own page. The merchant centre documentation had different requirements, but they have gone.

Reply:
@mattcoy-arcticleaf saying all that, I think the multiple offer solution is the best option at the moment as it causes price ranges.